docs/active_storage_migration.md

```diff
@@ -0,0 +1,387 @@
+# Active Storage migration plan
+
+## Contents
+
+* [Overview](#Overview)
+* [Stats Reports](#Stats-Reports)
+* [Messages](#Messages)
+* [Documents](#Documents)
+* [Clean up](#Clean-up)
+
+## Overview
+
+These notes outline the plan for migrating CCCD assets from Paperclip to
+Active Storage. Assets exist in three different places:
+
+* Stats reports
+* Attachments on messages
+* Evidence documents
+
+and each of these is dealt with separately. The process is similar for each but
+with some small differences and can be summarised as:
+
+1) Generate checksums for all existing assets. The Checksum is required by
+   Active Storage and is time consuming. However, it can be done in advance of
+   the migration to avoid disruption.
+2) Test that checksums have been created correctly.
+3) Copy information in the database about the assets from where Paperclip has
+   them to where they need to be for Active Storage.
+4) Test that document information has been copied correctly.
+5) Update the application to use Active Storage to create and access assets.
+6) Test that existing assets can be accessed and new assets can be created.
+
+## Stats Reports
+
+1) Generate checksums:
+
+```bash
+$ bundle exec rails 'storage:calculate_checksums[stats_reports]'
+```
+
+2) Ensure that the checksums have been created correctly:
+
+```ruby
+rails> srs = Stats::StatsReport.all
+
+# Total number of stats reports
+rails> srs.count
+
+# Number of stats reports with nil checksum
+rails> srs.where(as_attachment_checksum: nil).count
+# Should be zero
+
+# Number of stats reports with a valid checksum
+# A valid checksum ends with '==', such as '4+g5Nmjax5aCg3HtRmCO5Q=='
+rails> srs.select { |sr| sr.as_attachment_checksum&.match(/==$/) }.count
+# Should be the same as srs.count, above
+```
+
```

Comment:
don't you have a rake task that outputs missing checksums. be good to document that here

Reply:
and maybe extend that to perform the basic checksum validity check you have written here
```
srs.select { |sr| sr.as_attachment_checksum&.match(/==$/) }.count
```